cmd/olm/main.go

```diff
@@ -105,11 +115,16 @@ func main() {
 	}
 	defer operator.Cleanup()
 
+	mux := http.NewServeMux()
 	// Serve a health check.
-	http.HandleFunc("/healthz", func(w http.ResponseWriter, r *http.Request) {
+	mux.HandleFunc("/healthz", func(w http.ResponseWriter, r *http.Request) {
 		w.WriteHeader(http.StatusOK)
 	})
-	go http.ListenAndServe(":8080", nil)
+	// TODO: both of the following require vendor updates (add k8s.io/apiserver and update prometheus)
+	//healthz.InstallHandler(mux) //(less code)
+	//mux.Handle("/metrics", promhttp.Handler()) //other form is deprecated
+	mux.Handle("/metrics", prometheus.Handler())
+	go http.ListenAndServe(":8080", mux)
```

Comment:
do you know if there's a convention for this? is it common to serve metrics on the healthz port or do people define a separate port for each?

Reply:
Yeah, I think it's common to use the same port. Here's what the prometheus operator does (though they aren't using healthz here): https://github.com/coreos/prometheus-operator/blob/ec94db4149766312fb60a2d5fdc5aca3153386dd/cmd/operator/main.go#L201

Reply:
looks like you can do both without a new muxer? https://github.com/coreos/etcd-operator/blob/master/cmd/operator/main.go#L105-L106